lib/db/structs.proto

```diff
@@ -75,3 +76,14 @@ message CountsSet {
     repeated Counts counts  = 1  [(gogoproto.nullable) = false];
     int64           created = 2; // unix nanos
 }
+
+message ObservedFolder {
+    google.protobuf.Timestamp time  = 1 [(gogoproto.stdtime) = true];
```

Comment:
We use nanoseconds from `time.Unix` for times on metadata, though I don't really see the point of the nanos instead of just seconds :)

Reply:
In this case, seconds is perfectly enough IMO, as it just records the last connection attempt for user convenience.  So what other type should I use?

Reply:
int64 for unix seconds then.

Reply:
I like "speaking" type names instead of reusing basic types for differing semantics.  Using the standard `time.Time` (Go naming) has the advantage of fully handling formatting etc.  I'd really like to stick with the smart, obvious type.  Unless it's a major burden for some reason.

Storage size doesn't matter much since I hope nobody has millions of pending devices?

Reply:
I agree, storage is not a concern. However we depend on `google.protobuf.Timestamp` and `gogoproto.stdtime` not changing in future versions. Maybe that's a given, but if not I'd say it's worth the hassle to go to a type that definitely wont change (because adapting to library changes is annoying in general and time in specific is always hard).

Anyway this is not important at this stage of the PR, lets see what others think.

Reply:
@AudriusButkevicius @calmh Could you comment on the time issue please (`google.protobuf.Timestamp` vs `int64` unix seconds).

Reply:
I'm OK with this, though I'd drop a nullable=false in there to get a `time.Time` instead of `*time.Time`

Reply:
I came to the conclusion that anything in `google.protobuf` is meant to preserve protocol compatibility, which is the whole point of protobuf.  So it's highly unlikely to encounter breaking changes there.

Will change to `nullable=false` soon.